src/main/java/com/telefonica/iot/cygnus/interceptors/GroupingInterceptor.java

```diff
@@ -101,22 +142,55 @@ public void initialize() {
                     continue;
                 } // if
                 
-                String[] tokens = line.split("\\|");
-                
-                if (tokens.length < 5 || tokens[0].length() == 0 || tokens[1].length() == 0 || tokens[2].length() == 0
-                        || tokens[3].length() == 0 || tokens[4].length() == 0) {
-                    LOGGER.warn("Malformed matching rule, it will be discarded. Details=" + line);
-                    continue;
-                } // if
-                
-                int id = new Integer(tokens[0]);
-                ArrayList<String> fields = new ArrayList<String>(Arrays.asList(tokens[1].split(",")));
-                matchingTable.add(new MatchingRule(id, fields, tokens[2], tokens[3], tokens[4]));
+                jsonStr += line;
             } // while
+            
+            return jsonStr;
         } catch (IOException e) {
-            LOGGER.error("Runtime error (I/O exception. Details=" + e.getMessage() + ")");
-        } // try catch // try catch
-    } // initialize
+            LOGGER.error("Error while reading the Json-based grouping rules file. Details=" + e.getMessage() + ")");
+            return null;
+        } // try catch
+    } // readGroupingRulesFile
+    
+    private JSONArray parseGroupingRules(String jsonStr) {
+        if (jsonStr == null) {
+            return null;
+        } // if
+        
+        JSONParser jsonParser = new JSONParser();
+        
+        try {
+            return (JSONArray) ((JSONObject) jsonParser.parse(jsonStr)).get("grouping_rules");
+        } catch (ParseException e) {
+            LOGGER.error("Error while parsing the Json-based grouping rules file. Details=" + e.getMessage());
+            return null;
+        } // try catch
+    } // parseGroupingRules
+    
+    private LinkedList<GroupingRule> createGroupingRules(JSONArray jsonGroupingRules) {
+        if (jsonGroupingRules == null) {
+            return null;
+        } // if
+        
+        groupingRules = new LinkedList<GroupingRule>();
+        
+        for (Object jsonGroupingRule : jsonGroupingRules) {
+            JSONObject jsonRule = (JSONObject) jsonGroupingRule;
+            
+            if (isValid(jsonRule)) {
```

Comment:
Maybe some kind of logging message informing the user/developer/devop of a malformed rule would be appreciated to ease its detection (the consequence is that the rule is not applied and the user/developer/devop may have no idea the reason why in case of typos or alike :) )